tests/test_headers.py

```diff
@@ -190,7 +190,8 @@ def on_get(self, req, resp):
             append_link = (
                 resp.append_link if append_link is resp.add_link else resp.add_link
             )
-            append_link(*args, **kwargs)
+            with pytest.warns(DeprecatedWarning):
```

Comment:
This is not your fault, but it doesn't seem right.

My impression was that the above statement `append_link = resp.append_link if append_link is resp.add_link else resp.add_link` was meant to be reused for both `append_link()` and the now-deprecated `add_link()`, but your test asserts a warning indiscriminately, and the test passes :thinking: 

Could you investigate what is going on here? If we are only testing the deprecated method here, we ought to simplify the above statement. Otherwise, fix the logic.